test/e2e/apimachinery/resource_quota.go

```diff
@@ -2480,26 +2481,27 @@ func countResourceQuota(ctx context.Context, c clientset.Interface, namespace st
 	})
 }
 
-// wait for resource quota status to show the expected used resources value
+// Wait for resource quota status to show the expected used resources value.
+// Other resources are ignored.
 func waitForResourceQuota(ctx context.Context, c clientset.Interface, ns, quotaName string, used v1.ResourceList) error {
-	return wait.PollUntilContextTimeout(ctx, framework.Poll, resourceQuotaTimeout, false, func(ctx context.Context) (bool, error) {
-		resourceQuota, err := c.CoreV1().ResourceQuotas(ns).Get(ctx, quotaName, metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
+	// The template emits the actual ResourceQuota object as YAML.
+	// In particular the ManagedFields are interesting because both
+	// kube-apiserver and kube-controller-manager set the status.
+	haveUsedResources := gcustom.MakeMatcher(func(resourceQuota *v1.ResourceQuota) (bool, error) {
 		// used may not yet be calculated
 		if resourceQuota.Status.Used == nil {
 			return false, nil
 		}
 		// verify that the quota shows the expected used resource values
 		for k, v := range used {
 			if actualValue, found := resourceQuota.Status.Used[k]; !found || (actualValue.Cmp(v) != 0) {
-				framework.Logf("resource %s, expected %s, actual %s", k, v.String(), actualValue.String())
 				return false, nil
 			}
 		}
 		return true, nil
-	})
+	}).WithTemplate("Expected:\n{{.FormattedActual}}\n{{.To}} have the following .status.used entries:\n{{range $key, $value := .Data}}    {{$key}}: \"{{$value.ToUnstructured}}\"\n{{end}}").WithTemplateData(used /* Formatting of the map is done inside the template. */)
```

Comment:
trying to see if I get this right 
- `Data` will be the var `used` . that is a list, inside it is iterated and dump as unstructured?
- `FormattedActual ` is `c.CoreV1().ResourceQuotas(ns).Get, quotaName, metav1.GetOptions{})` 

is that right?

Reply:
recognize the representation is much better, but it requires to learn a lot of internals of gomega, to understand it 

/lgtm

Reply:
> Data will be the var used

Correct. This is standard Go templating. The only Gomega specific aspect is that it provides the input (`.Actual`, `.FormattedActual`, `.Data`) and some helpers (`format`).

> FormattedActual is `c.CoreV1().ResourceQuotas(ns).Get, quotaName, metav1.GetOptions{})`

Yes, after applying the Gomega formatter.